std/os/linux.zig

```diff
@@ -907,6 +907,50 @@ pub fn io_uring_register(fd: i32, opcode: u32, arg: ?*const c_void, nr_args: u32
     return syscall4(SYS_io_uring_register, @bitCast(usize, isize(fd)), opcode, @ptrToInt(arg), nr_args);
 }
 
+pub fn fcntl(fd: i32, cmd: u32, arg: u32) usize {
```

Comment:
Since there is an equivalent `fcntl` function in C that does not do any hidden "magic" that you're doing here, I highly discourage you from changing the semantics of `fcntl` in zig.  Rather, you should keep `fcntl` just as it works in C, and if you want to create some magic around it, then call it a different name.